agent/dns.go

```diff
@@ -413,7 +413,9 @@ func (d *DNSServer) handlePtr(resp dns.ResponseWriter, req *dns.Msg) {
 	m.SetReply(req)
 	m.Compress = !cfg.DisableCompression
 	m.Authoritative = true
-	m.RecursionAvailable = (len(cfg.Recursors) > 0)
+	recursorsEnabled := atomic.LoadUint32(&(d.recursorEnabled)) != 1
+	recursionAvailable := recursorsEnabled && len(cfg.Recursors) > 0
```

Comment:
~ For a non-configured fast path, I think we could skip the atomic load if we condensed to
```suggestion
	recursionAvailable := len(cfg.Recursors) > 0 && atomic.LoadUint32(&(d.recursorEnabled)) != 1
```
Not sure whether the compiler will inline that automatically anyway.